### PR TITLE
Fix: apiserver image build and add CI check for it

### PR DESCRIPTION
**Pull request by wonderflow**
Signed-off-by: Jianbo Sun <jianbo.sjb@alibaba-inc.com>

### Description of your changes

Fixes https://github.com/kubevela/kubevela/actions/runs/3537032744/jobs/5936644544

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -173,3 +173,35 @@ jobs:
         run: |
           move .\bin\vela .\bin\vela.exe
           .\bin\vela.exe version
+
+    image-build-check:
```
Comment (barnettZQG) [inline .github/workflows/go.yml:177]: Is this a repeat job? There is a `Registry` workflow to check building images.

Comment (wonderflow) [reply .github/workflows/go.yml:177]: The Registry CI only works in release/master branch, I'm making it to be PR normal check.

Comment (barnettZQG) [reply .github/workflows/go.yml:177]: We could set the Registry workflow to work in all branches?

Comment (wonderflow) [reply .github/workflows/go.yml:177]: That is a more complicate CI and contains secrets, we don't need to do that.

**Review by barnettZQG — APPROVED** [review]

**Pull request merged**